sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -290,6 +304,13 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 				registryAddress, secret.GetNamespace(), secret.GetName())
 		}
 
+		if err := validateSecret(registryAddr, dce); err != nil {
```

Comment:
Should we validate configs before using them so right after `getDockerConfigFromSecret` remove invalid configs and then check the len.

Reply:
Yes, the second idea is something that would work and makes sense.
Validation right after `getDockerConfigFromSecret` would require to iterate over the registries only to do the validation, so it would be similar to current approach.

I will give it a try to validate in `config.go`.

Reply:
However, in `func decodeDockerConfigFieldAuth(field string) (username, password string, err error) {`, we only decode username, password and auth string. The registry address is missing. We would need to validate in both places to cover everything (which is maybe not so bad).

Reply:
I tried it and there is one more problem with that approach: because of this https://github.com/stackrox/stackrox/blob/9da1da2bf5a0d3287f1942f0566a5556f2b658e1/sensor/kubernetes/listener/resources/secrets.go#L225-L228, we would reject the entire secret if at least one registry contains invalid data. If that secret contains more registries, we would loose them. WDYT?

Reply:
> The registry address is missing. 

Is it possible to have non UTF-8 character in address? Maybe this should be validated by k8s I reported it.
 - https://github.com/kubernetes/kubernetes/issues/131982
 
 
 > we would reject the entire secret if at least one registry contains invalid data. If that secret contains more registries, we would loose them. WDYT?
 
 I we can inform User about it I think it will be fine but with current implementation it's only logged in sensor and this error will not be passed to central.

Reply:
It should not be possible, but I cannot exclude some exotic scenarios, where base64 encoded secrets are copy-pasted manually and some data may get corrupted unintentionally. I find checking that really cheap, so maybe we can keep it?